src/main/java/de/tum/in/www1/exerciseapp/domain/DragAndDropQuestionStatistic.java

```diff
@@ -106,7 +167,48 @@ public void addResult(SubmittedAnswer submittedAnswer, boolean rated) {
      */
     @Override
     public void removeOldResult(SubmittedAnswer submittedAnswer, boolean rated) {
-        //TO-DO: Moritz Issig
+
+        if(submittedAnswer == null) {
+            return;
+        }
+
+        DragAndDropSubmittedAnswer ddSubmittedAnswer = (DragAndDropSubmittedAnswer)submittedAnswer;
+
+        if(rated) {
+            //decrease the rated participants
+            setParticipantsRated(getParticipantsRated() - 1);
+
+            if(ddSubmittedAnswer.getMappings() != null) {
+                // decrease rated dropLocationCounter if dropLocation is correct
+                for (DropLocationCounter dropLocationCounter: dropLocationCounters) {
+                    if(dropLocationCounter.getDropLocation().isDropLocationCorrect(ddSubmittedAnswer.getMappings())) {
+                        dropLocationCounter.setRatedCounter(dropLocationCounter.getRatedCounter() - 1);
+                    }
+                }
+            }
+            // decrease rated correctCounter if answer is complete correct
+            if(getQuestion().isAnswerCorrect(ddSubmittedAnswer)) {
+                setRatedCorrectCounter(getRatedCorrectCounter() - 1);
+            }
+        }
+        // Result is unrated
+        else{
+            //decrease the unrated participants
+            setParticipantsRated(getParticipantsUnrated() - 1);
```

Comment:
same problem: This line seems to be wrong as you are in the unrated branch but you call "setParticipantsRated(...)"

--> It is obvious that you just copied and pasted the source code, which is really really bad. Please avoid this and make the code reusable independent of the + and -
This would be possible, if you e.g. create a new method with a parameter changeStatisticBy(int value) and you either pass +1 or -1